channel/adjudicator.go

```diff
@@ -25,41 +25,76 @@ import (
 )
 
 type (
-	// An Adjudicator represents an adjudicator contract on the blockchain. It has
-	// methods for state registration, progression and withdrawal of channel
-	// funds. A channel state needs to be registered before it can be progressed
-	// or withdrawn. No-App channels skip the force-execution phase and can be
-	// withdrawn directly after the refutation phase has finished. Channels in a
-	// final state can directly be withdrawn after registration.
+
+	// Adjudicator is the interface that groups the Register, Withdraw,
+	// Progress and Subscribe methods.
+	//
+	// An adjudicator represents an adjudicator contract on the blockchain. A
+	// channel state needs to be registered before it can be progressed or
+	// withdrawn. No-App channels skip the force-execution phase and can be
+	// withdrawn directly after the refutation phase has finished. Channels in
+	// a final state can directly be withdrawn after registration.
 	//
 	// Furthermore, an Adjudicator has a method for subscribing to
 	// AdjudicatorEvents. Those events might be triggered by a Register or
 	// Progress call on the adjudicator from any channel participant.
 	Adjudicator interface {
-		// Register should register the given ledger channel state on-chain.
-		// If the channel has locked funds into sub-channels, the corresponding
-		// signed sub-channel states must be provided.
+		Registerer
```

Comment:
I see two aspects to this discussion:

1. Using `Adjudicator` interface in its current form in the `Watcher`.

It is a good practice to require only the smallest possible interface in a function argument. Because, this makes the dependencies of the function explicit.

2. Over time, we might add functionalities like `Progressing` and `Concluding` to the watcher. 

Looking at the [proposal](https://github.com/hyperledger-labs/perun-proposals/blob/main/design/003-IoT-Adoption.md) for adopting perun-framework for IoT use cases, I don't see that watcher will be extended to support these functionalities. On the other hand, this functionalities (Progressing, Concluding) might be implemented by the `Funding` component (shown in the figure 2 in the proposal).

3. Split the methods required only for the watcher.

In that case, the definition of `Adjudicator` interface will be a mix of methods and interfaces. See below:

```
type Adjudicator interface {
    Progress(..)
    Withdraw(...)
    RegisterSubscriber
}

type RegisterSubscriber interface {
    Register(...)
    Subscriber(...{
}
```

This would mean documentation for some methods of the interface would be on the `Adjudicator` type and for the remaining interfaces on the `RegisterSubscriber` type. I prefer to avoid this by defining interfaces for every method and composing the `Adjudicator` from them. But, on this part I wouldn't mind do it as described the above snippet.

Reply:
OK, I think you have convinced me that we can split the interface up 👍